internal/generator/vector/filter/prune/prune.vrl.tmpl

```diff
@@ -0,0 +1,22 @@
+{{- if .NotIn}}
+notIn = {{.NotIn}}
+
+# Prune keys not in `notIn` list
+new_object = {}
+for_each(notIn) -> |_index, pathSeg| {
+    val = get(., pathSeg) ?? null
```

Comment:
can we use `exist` function instead of `get` https://vector.dev/docs/reference/vrl/functions/#exists ?

Reply:
I am using `get` here because I broke up the field path into an array of path segments. The `get` function expects an array instead of a path. I could not use something like:
```
paths = [.log, .message]
```
and iterate through that because `for_each` would only extract the values and not preserve the path for the `exists` function.